lang/semgrep-grammars/src/semgrep-julia/grammar.js

```diff
@@ -20,6 +20,28 @@ module.exports = grammar(base_grammar, {
   rules: {
     semgrep_ellipsis: $ => '...',
 
+    // Metavariables
+    identifier: ($, previous) => {
+      return token(
+        choice(
+          previous,
+          // We allow an identifier to be the same as an identifier, but prefixed
+          // with a dollar sign.
+          // We could use a more specific regular expression, which would match
```

Comment:
I would vote for a better regexp with $[A-Z]+
at a word boundary.

Reply:
@aryx Little bit of an issue:
<img width="1118" alt="image" src="https://github.com/returntocorp/ocaml-tree-sitter-semgrep/assets/49291449/89457a21-af68-4f0e-8f04-f5034117860e">

https://github.com/tree-sitter/tree-sitter/blob/62e96c9f61af086641699a8848cf881d27e9afdf/cli/src/generate/prepare_grammar/expand_tokens.rs#L255

It seems that `tree-sitter` does not support word boundary assertions.

Reply:
@aryx Some more problems:
`identifier` in the Julia grammar is what is known as the "word token", which is a special hard-coded token that is meant to disambiguate identifiers which may also be keywords (see https://tree-sitter.github.io/tree-sitter/creating-parsers#keyword-extraction)

 Notably, that token is not allowed to be a non-terminal at the rule level. It can only be a single token.

This might seem OK, because tokens themselves can be `choice`s of other token elements. The problem is that they cannot be `Symbol`s, which includes external symbols. This means I cannot make `identifier` the union of the old `identifier` and an external token, which was how I thought I might fix this. I can make `identifier` itself external, but that means I would have to re-implement it in C, which is a very complicated regex.

In summary, this is a really hard grammar to hack.